app/controllers/application_controller.rb

```diff
@@ -37,8 +37,12 @@ def routing_error
       when Common::Exceptions::BaseError
         exception
       when Common::Client::Errors::BackendServiceError
-        additional_attributes = { source: exception.developer_message, meta: exception.as_json }
-        Common::Exceptions::ClientError.new(exception.message, additional_attributes)
+        additional_attributes = {
+          detail: exception.message,
+          source: exception.developer_message,
+          meta: exception.as_json
+        }
+        Common::Exceptions::ClientError.new(nil, additional_attributes)
```

Comment:
Yes, I can rename it. Would Common::Exceptions::BackendServiceError be more appropriate? 

I think I'd prefer to keep client exceptions distinct from common exceptions for now. The client has its own distinct responsibilities, and I don't think it should be concerned with having the capability of rendering according to JSONAPI spec. I think that is a distinctly separate responsibility that Common::Exceptions should handle.

I think I agree with you though. To me the responsibilities for everything under Client are distinct from Common::Exceptions. They don't need to be, but I think as it stands now, were these to be packaged as a gem, I would consider them as 2 separate gems.

Hope that helps to clarify. Let me know if it doesn't.